### PR TITLE
[mdatagen] Generate documentation for metrics enabling/disabling

### PR DESCRIPTION
**Pull request by dmitryax**
The new metrics builder introduced an ability to add optional metrics to scrapers and a settings option to disable/enable particular metrics. This change adds automatically generated documentation for this feature.

Tracking issue: https://github.com/open-telemetry/opentelemetry-collector/issues/10904

### CODE REVIEW

**Comment by dmitryax** [comment]
Metrics enabled by default are highlighted in bold in the table. There are maybe other options, for example we can add another column instead of highlighting the names. Not sure about the column header in that case, something like "Enabled by default?" looks too long for Yes/No values. I'm open to suggestions.

cc @djaglowski @bogdandrutu

**Comment by bogdandrutu** [comment]
Needs a rebase

**Comment by dmitryax** [comment]
Rebased

File: cmd/mdatagen/documentation.tmpl [hunk 1]
```diff
@@ -9,10 +9,22 @@ These are the metrics available for this scraper.
 | Name | Description | Unit | Type | Attributes |
 | ---- | ----------- | ---- | ---- | ---------- |
 {{- range $metricName, $metricInfo := .Metrics }}
-| {{ $metricName }} | {{ $metricInfo.Description }}{{ if $metricInfo.ExtendedDocumentation }} {{ $metricInfo.ExtendedDocumentation }}{{ end }} | {{ $metricInfo.Unit }} | {{ $metricInfo.Data.Type }}
+| {{ if and $.ExpGen $metricInfo.IsEnabled }}**{{ end }}{{ $metricName }}{{ if and $.ExpGen $metricInfo.IsEnabled }}**
```
Comment (bogdandrutu) [inline cmd/mdatagen/documentation.tmpl:12]: Confused about the need of ExpGen. In the old format all metrics are enabled correct? So why do you need this?

Comment (dmitryax) [reply cmd/mdatagen/documentation.tmpl:12]: They are enabled, but there is no way to reconfigure it, so I left it as is to not bring confusion

File: cmd/mdatagen/loader.go [hunk 2]
```diff
@@ -110,8 +114,8 @@ type templateContext struct {
 	metadata
 	// Package name for generated code.
 	Package string
-	// ExpFileNote contains a note about experimental metrics builder.
-	ExpFileNote string
+	// ExpGen identifies whether the experimental metrics generator is used.
+	ExpGen bool
```
[the remark this answers is not in the document]
Comment (dmitryax) [reply cmd/mdatagen/loader.go]: Added a TODO

File: cmd/mdatagen/documentation.tmpl (hunk 1 above)
Comment (dmitryax) [reply cmd/mdatagen/documentation.tmpl]: Ok, tried that. All metrics from the old generators are highlighted now but with no info about optional metrics. Let me know how does it look now

**Review by dashpole — APPROVED** [review]

**Review by djaglowski — APPROVED** [review]

**Pull request merged**